dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -365,7 +404,14 @@ private DependencyGraphResult levelOrder(
       Stack<DependencyNode> parentNodes, DependencyNode failedDependencyNode) {
     List<DependencyNode> fullPath = new ArrayList<>();
     fullPath.addAll(parentNodes);
-    fullPath.add(failedDependencyNode);
```

Comment:
testBuildLinkageCheckDependencyGraph_catchRootException revealed that sometimes there was duplicate element in dependency path. For example:

```
xerces:xerces-impl:jar:2.6.2 was not resolved. Dependency path: ant:ant:jar:1.6.2 (compile) > xerces:xerces-impl:jar:2.6.2 (compile?) > xerces:xerces-impl:jar:2.6.2 (compile?)
```